visualize-admin/visualization-tool
# fix: AMDP cubes temporal dimension

Closes #1234

This PR brings `TemporalEntityDimension` to Visualize, so that AMDP cubes can be properly filtered and displayed (so it's possible to use line or area charts, animations, etc).

Currently these are the only cubes that use temporal entities (https://int.ld.admin.ch/time/month/2004-09) that are available on LINDAS.

As we are not aware of other types of temporal entities (e.g. semester or second) and don't know which properties they will specify, a decision was made to rely on `schema:position`, which holds the formatted dates, to retrieve them in a generic way. A [proposal was made](https://zulip.zazuko.com/#narrow/stream/40-bafu-ext/topic/temporal.20entity.20and.20schema.3AsameAs/near/402718) to use an "agreed on" property – `schema:sameAs` – that we could safely rely on to retrieve the dates without a complex overhead logic.

### How to test
1. Go to [this link](https://visualization-tool-git-feat-bring-amdp-cubes-to-visualize-ixt1.vercel.app/en/create/new?cube=https://agriculture.ld.admin.ch/foag/cube/MilkDairyProducts/Consumption_Price_Month&dataSource=Int).
2. See that the column chart is sorted correctly.
3. Open `Horizontal Axis`.
4. See that you can filter the range.
5. See that it's possible to change the chart type to Lines or Areas.
6. Select a different dimension as X axis.
7. See that it's possible to use Date an Animation field.
